libraries/helpers.rb

```diff
@@ -619,3 +630,18 @@ def execute_command(command, user = "root", timeout = 300, raise_on_error = true
   raise_command_error(command, cmd) if raise_on_error && cmd.error?
   cmd.stdout.strip
 end
+
+# load cluster configuration file into node object
+def load_shared_storages_mapping
+  ruby_block "load shared storages mapping during cluster update" do
+    block do
+      require 'yaml'
+      # regenerate the share storages mapping file after update
+      shared_storages_mapping = YAML.safe_load(File.read(node['cluster']['shared_storages_mapping_path']))
+      update_shared_storages_mapping = YAML.safe_load(File.read(node['cluster']['update_shared_storages_mapping_path']))
```

Comment:
What about assigning the parsed mapping directly instead of declaring additional variables?

I mean, going from this:

```
shared_storages_mapping = YAML.safe_load(File.read(node['cluster']['shared_storages_mapping_path']))
node.default['cluster']['shared_storages_mapping'] = shared_storages_mapping
```

to this:

```
node.default['cluster']['shared_storages_mapping'] = YAML.safe_load(File.read(node['cluster']['shared_storages_mapping_path']))
```

Reply:
Done